storage/store.go

```diff
@@ -1822,3 +1822,21 @@ func (s *Store) PublishStatus() error {
 func (s *Store) SetRangeRetryOptions(ro retry.Options) {
 	s.ctx.RangeRetryOptions = ro
 }
+
+// hasReplicaExist used to judgement if "rngDesc" has overlap with ranges
+// in "replicasByKey" btree, it is not thread saft.
+func (s *Store) hasOverlapReplica(rngDesc *roachpb.RangeDescriptor) bool {
+	var rng *Replica
+	s.replicasByKey.AscendGreaterOrEqual((rangeBTreeKey)(rngDesc.StartKey.Next()), func(i btree.Item) bool {
+		rng = i.(*Replica)
+		return false
+	})
+
+	if rng != nil {
```

Comment:
everything below this line is equal to

`return rng != nil && rng.Desc().StartKey.Less(rngDesc.EndKey)`